src/cli/command_processor.py

```diff
@@ -14,71 +14,105 @@ def __init__(self, remote_api: RemoteApi, local_api: LocalApi):
 
     @log_function
     def login(self, host: str, username: str, password: str) -> None:
-        pass
+        self.__remote.login(host, username, password)
 
     @log_function
     def logout(self, host: str) -> None:
-        pass
+        self.__remote.logout(host)
 
     @log_function
     def applications_create(
         self, application: str, roles: List[str], path: Path
     ) -> None:
-        pass
+        self.__local.create_application(application, roles, path)
 
     @log_function
     def applications_init(self, path: Path) -> None:
-        pass
+        self.__local.init_application(path)
 
     @log_function
     def applications_upload(self, application: str) -> None:
-        pass
+        self.__remote.upload_application(application)
 
     @log_function
     def applications_list(self, remote: bool, local: bool) -> List[ApplicationType]:
-        return []
+        app_list = []
+        if remote:
+            app_list.extend(self.__remote.list_applications())
+
+        if local:
+            app_list.extend(self.__local.list_applications())
+
+        return app_list
 
     @log_function
     def applications_delete(self, application: str) -> None:
-        pass
+        self.__remote.delete_application(application)
 
     @log_function
     def applications_publish(self, application: str) -> None:
-        pass
+        self.__remote.publish_application(application)
 
     @log_function
     def applications_validate(self, application: str) -> None:
-        pass
+        self.__local.is_application_valid(application)
 
     @log_function
     def experiments_create(self, name: str, application: str, local: bool) -> None:
-        pass
+        if local:
+            self.__local.get_application_config(application)
```

Comment:
Why is it enough to get application config to create an experiment? I think this is an error in the sequence diagram (copy error) It should have called create_experiment with the app config.